analyze/analyze.rs

```diff
@@ -801,3 +799,84 @@ pub fn diff(
     let diff = Diff { deltas };
     Ok(Box::new(diff) as Box<traits::Emit>)
 }
+
+#[derive(Debug)]
+struct Garbage {
+    items: Vec<ir::Id>,
+    opts: opt::Garbage,
+}
+
+impl traits::Emit for Garbage {
+    fn emit_text(&self, items: &ir::Items, dest: &mut io::Write) -> Result<(), traits::Error> {
+        let mut table = Table::with_header(vec![
+            (Align::Right, "Bytes".to_string()),
+            (Align::Right, "Size %".to_string()),
+            (Align::Right, "Item".to_string()),
+        ]);
+
+        for &id in &self.items {
+            let item = &items[id];
+            let size = item.size();
+            let size_percent = (f64::from(size)) / (f64::from(items.size())) * 100.0;
+            table.add_row(vec![
+                size.to_string(),
+                format!("{:.2}%", size_percent),
+                item.name().to_string(),
+            ]);
+        }
+
+        write!(dest, "{}", &table)?;
+        Ok(())
+    }
+
+    fn emit_json(&self, items: &ir::Items, dest: &mut io::Write) -> Result<(), traits::Error> {
+        let mut arr = json::array(dest)?;
+
+        for &id in &self.items {
+            let item = &items[id];
+
+            let mut obj = arr.object()?;
+            obj.field("name", item.name())?;
+
+            let size = item.size();
+            let size_percent = (f64::from(size)) / (f64::from(items.size())) * 100.0;
+            obj.field("bytes", size)?;
+            obj.field("size_percent", size_percent)?;
+        }
+
+        Ok(())
+    }
+}
+
+/// Find items that are not transitively referenced by any exports or public functions.
+pub fn garbage(
+    _items: &ir::Items,
+    _opts: &opt::Garbage,
```

Comment:
Variables with a leading underscore are usually reserved for unused variables. Stylistically, it is a little weird for these parameters to be used despite the leading underscore. These should be changed to plain `items` and `opts`.

Reply:
Oh shoot! I meant to change those, I had placed the garbage function declaration in with an `unimplemented` macro when I was working on the plumbing elsewhere. Will do!